plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -59,11 +56,10 @@
         implements UpdatablePageSource
 {
     private final Schema schema;
-    private final Type[] columnTypes;
     private final int[] expectedColumnIndexes;
     private final ConnectorPageSource delegate;
     private final Optional<ReaderProjectionsAdapter> projectionsAdapter;
-    private final Optional<DeleteFilter<TrinoRow>> deleteFilter;
+    private final Optional<RowPredicate> deletePredicate;
```

Comment:
We probably need to track memory contained in this field to prevent worker node crashes.
See https://github.com/trinodb/trino/pull/9914
cc @losipiuk

Reply:
Agreed, though we aren't doing that today, since this isn't a regression. We can do that as a follow up. I didn't see an easy way to get memory usage from RoaringBitmap. We could serialize and use that as an estimate, or do a simple estimate based on the cardinality (multiply by some constant factor).